x/pep/keeper/msg_register_contract.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k msgServer) RegisterContract(
+	goCtx context.Context,
+	msg *types.MsgRegisterContract,
+) (*types.MsgRegisterContractResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	var contDetails = types.ContractDetails{
+		Registrar:       msg.Creator,
+		ContractAddress: msg.ContractAddress,
+	}
+
+	entry, found := k.GetContractEntriesByID(ctx, msg.Identity)
+	if found {
+		if len(entry.Contracts) != 0 {
+			for _, c := range entry.Contracts {
+				if c.ContractAddress == msg.ContractAddress {
+					return nil, errors.New("contract is already registered for this identity")
+				}
+			}
+		} else {
+			entry.Contracts = make([]*types.ContractDetails, 0)
```

Comment:
Don't think this is needed ?

Reply:
remember the chain crashed last time whenever a general keyshare request was made. and that was because a particular slice had not been made and as a result there was a nil pointer.

i just am being overly safe here not to repeat another chain crash because of a malformed request or something. so, I made doubly sure by initializing a blank slice

Reply:
If that's the case isn't this suppose to be executed if the entry not found ?

```
entry, found := k.GetContractEntriesByID(ctx, msg.Identity)
if found {
    ....
} else {
    entry.Contracts = make([]*types.ContractDetails, 0)
}
```

If `len(entry.Contracts) == 0` then I guess it is already an empty array ?